client/asset/eth/fundingcoin.go

```diff
@@ -0,0 +1,70 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"decred.org/dcrdex/dex/encode"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const fundingCoinIDSize = 36 // address (20) + amount (8) + nonce (8) = 36
+
+// FundingCoinID is an identifier for a coin which has not yet been sent to the
+// swap contract.
+type FundingCoinID struct {
+	Address common.Address
+	Amount  uint64
+	Nonce   [8]byte
```

Comment:
I actually thought we were going to get rid of the `Nonce`. Do you see value in it?

Reply:
I think we can eliminate it, but with the present client-side coin locking with these as map keys, it has to stay for now.
I'd like another pr to revisit the client's funding coin locking, but I don't think it's critical.  I had closed the one I had previously started for discussion.
This pr at least isolates this funding coin notion to client packages.

Reply:
Sorry, I should have pointed out in the PR description that this type is essentially the old `AmountCoinID` from the now removed cointypes.go, just with the coin type flag removed.

Reply:
I can remove the nonce in #1325. The Refund function will not receive the funding coin ids used to reserve funds to use for gas in case of a refund, so the locked funds map will need to be removed and replaced by a uint64 as @chappjc had done in #1289.